git-compat-util.h

```diff
@@ -430,6 +430,20 @@ char *gitdirname(char *);
 #define NI_MAXSERV 32
 #endif
 
+/* PATH_MAX is inherited from mingw64/include/limits.h, with a value of 260.
+ * The problem is that even after enabling Windows long path support and
+ * 'core.longpaths' there are still some Git operations that will fail, such
+ * as cloning a submodule (see setup_explicit_git_dir).
+ *
+ * I've observed some CI runners that have temp clones in a deeply nested
+ * path which will break even some medium sized repository paths.
+ *
+ * Redefining this macro addresses that issue.
+ */
+#if defined(_WIN32) || defined(__CYGWIN__) || defined(__MINGW32__)
```

Comment:
I can understand most of this, but since you are contributing this to the Git for Windows project, I don't think it would be appropriate to include the `_CYGWIN` part.

Also, the commit message is a bit too mum about rationale. Please follow the guidance in https://github.blog/2022-06-30-write-better-commits-build-better-projects/ to improve it, in particular with a strong focus on this part:

  | What you’re doing | Why you’re doing it
-- | -- | --
High-level (strategic) | **Intent** (what does this accomplish?) | **Context** (why does the code do what it does now?)
Low-level (tactical) | **Implementation** (what did you do to accomplish your goal?) | **Justification** (why is this change being made?)